docs/rfcs/1013-parsec-account.md

```diff
@@ -0,0 +1,456 @@
+{
+  "cmd": "account_send_email_validation_token",
+  "req": {
+    "email": "EmailStr"
+  },
+  "reps": [
+    {
+      "status": "ok"
+    },
+    {
+      "status": "invalid_email"
+    }
+  ]
+}
+```
+
+On `ok`, the server would have sent a mail with a unique token used for next the request used to register the authentication method.
+If the email is already registered in the service, a `ok` response will still be sent (without sending an email)
+to avoid creating an oracle about emails registered in the service.
+
+```yml
+{
+  "cmd": "account_password_create",
+  "req": {
+    "email_validation_token": "EmailValidationToken",
+    "account_label": "String",
+    "auth_method_key_params": "KeyAlgorithm",
+    "auth_method_hmac_key": "HMACKey",
+    "encrypted_vault_key": "EncryptedSymKey"
+  },
+  "reps": [
+    {
+      "status": "ok",
+    },
+    {
+      "status": "invalid_email_validation_token"
+    }
+  ]
+}
+```
+
+Upon receiving the request, the server will create the [`Account`], [`AuthMethod`], and [`Vault`]
+
+#### Uploading a new device
+
+```yaml
+{
+  "cmd": "protected_device_upload",
+  "req": {
+    "device": "UploadedDevice"
+  },
+  "reps": {
+    "ok": {},
+    # The server compare the human handle of the device to the email of parsec-account
+    "invalid_human_handle": {}
+  }
+}
+```
+
+```yaml
+{
+  "label": "UploadedDevice",
+  "type": "uploaded_device",
+  "other_fields": {
+    "organization_id": "OrganizationID",
+    "device_id": "DeviceID",
+    "encrypted_device": "EncryptedDevice"
+  }
+}
+```
+
+#### List devices
+
+To list the device linked to the authentication method's vault:
+
+```yaml
+{
+  "cmd": "protected_device_list",
+  "req": {},
+  "reps": {
+    "ok": {
+      "devices": "Vec<DeviceInfo>"
+    }
+  }
+}
+```
+
+`DeviceInfo` is an aggregation of [`ProtectedDevice`] and additional information retrieved from the metadata server.
+
+```yaml
+{
+  "label": "DeviceInfo",
+  "type": "device_info",
+  "other_fields": {
+    "organization_id": "OrganizationID",
+    "device_id": "DeviceID",
+    "user_id": "UserID",
+    "HumanHandle": "HumanHandle",
+    "created_on": "Timestamp",
+    "revoked_on": "Option<Timestamp>",
+    "encrypted_device": "EncryptedDevice"
+  }
+}
+```
+
+#### Recovery device list
+
+To list all protected device from all vault:
+
+```yaml
+{
+  "cmd": "protected_device_recovery_list",
+  "req": {},
+  "reps": {
+    "ok": {
+      "vaults": "Vec<VaultInfo>",
+      "devices": "Vec<DeviceInfo>"
+    }
+  },
+  "custom_types": {
+    "VaultInfo": {
+      "id": "VaultID",
+      "created_on": "Timestamp",
+      "created_by": "Creator",
+      "auth_methods": "Vec<AuthMethod>"
+    },
+    "AuthMethod": {
+      "algorithm": "KeyAlgorithm",
+      "created_on": "Timestamp",
+      "created_by": "Creator"
+    }
+  }
+}
+```
+
+> [!CAUTION]
+> The returned list does not include device from the current vault (since we already have access to them and do not need to recover them)
```

Comment:
I wouldn't put this in a CAUTION, but as part of the description of the route

```suggestion
The returned list does not include devices from the current vault (since user already has access to them and do not need to recover them). Only the vaults that still contain to-be-recovered protected devices are listed.

Recovery can be done in two ways:

- By remembering the old authentication method's password (since vaults are not removed)
- By recovering the Vault key. For this purpose, we should consider providing a way to export the vault key.
```